feeds/wifi-trunk/hostapd/patches/901-hapd-ubus-channel-switch.patch

```diff
@@ -0,0 +1,185 @@
+Index: hostapd-2020-06-08-5a8b3662/src/ap/dfs.c
+===================================================================
+--- hostapd-2020-06-08-5a8b3662.orig/src/ap/dfs.c
++++ hostapd-2020-06-08-5a8b3662/src/ap/dfs.c
+@@ -18,7 +18,7 @@
+ #include "drivers/driver.h"
+ #include "dfs.h"
+ #include "crypto/crypto.h"
+-
++#include "ubus.h"
+ 
+ static int dfs_get_used_n_chans(struct hostapd_iface *iface, int *seg1)
+ {
+@@ -1171,6 +1171,7 @@ static int hostapd_dfs_start_channel_swi
+ 	/* Channel configuration will be updated once CSA completes and
+ 	 * ch_switch_notify event is received */
+ 
++	hostapd_ubus_handle_channel_switch_event(iface, HOSTAPD_UBUS_DFS_SWITCH,  channel->freq);
+ 	wpa_printf(MSG_DEBUG, "DFS waiting channel switch event");
+ 	return 0;
+ }
+Index: hostapd-2020-06-08-5a8b3662/src/ap/ubus.c
+===================================================================
+--- hostapd-2020-06-08-5a8b3662.orig/src/ap/ubus.c
++++ hostapd-2020-06-08-5a8b3662/src/ap/ubus.c
+@@ -29,6 +29,8 @@ static int ctx_ref;
+ static char** bss_lst = NULL;
+ static size_t bss_nr = 0;
+ 
++static struct hostapd_chan_event_list *chan_events;
++
+ static inline struct hapd_interfaces *get_hapd_interfaces_from_object(struct ubus_object *obj)
+ {
+ 	return container_of(obj, struct hapd_interfaces, ubus);
+@@ -709,6 +711,44 @@ static int hostapd_sessions(struct ubus_
+ 	return 0;
+ }
+ 
++static int hostapd_get_chan_switch_events(struct ubus_context *ctx,
++			struct ubus_object *obj,
++		        struct ubus_request_data *req,
++			const char *method,
++                        struct blob_attr *msg)
++{
++	void *a = NULL;
++	void *t = NULL;
++
++
++	struct hostapd_chan_event_list *entry;
++	blob_buf_init(&b_ev, 0);
++	a = blobmsg_open_table(&b_ev, "chan_switch_event");
++	list_for_each_entry(entry, &chan_events->list, list)
++	{
++		t = blobmsg_open_table(&b_ev, "event");
++		blobmsg_add_u32(&b_ev, "radio_name", entry->records.band);
++		blobmsg_add_u8(&b_ev, "reason", entry->records.reason);
++		blobmsg_add_u64(&b_ev, "timestamp", entry->records.ts.tv_sec * (uint64_t)1000000);
```

Comment:
This way we will not be able to achieve the targeted milliseconds or microseconds precision. tv_nsec (which holds the rest of the elapsed time in nanoseconds) needs to be used in addition to tv_sec ofcourse with the desired ms/us conversion.